src/repo.rs

```diff
@@ -1891,6 +1891,42 @@ impl Repository {
             Ok(())
         }
     }
+
+    /// Add an ignore rule for a repository.
+    pub fn add_ignore_rule(&self, rule: &str) -> Result<(), Error> {
+        let rules = CString::new(rule)?;
+        unsafe {
+            try_call!(raw::git_ignore_add_rule(self.raw, rules));
+        }
+        Ok(())
+    }
+
+    /// Add ignore rules for a repository.
+    pub fn add_ignore_rules(&self, rules: &Vec<&str>) -> Result<(), Error> {
```

Comment:
In general this library tries to match upstream libgit2 as much as possible, so perhaps this could be omitted for now?